go/rpcclientlib/viewing_key_client.go

```diff
@@ -181,13 +187,14 @@ func (c *ViewingKeyClient) Stop() {
 
 func (c *ViewingKeyClient) SetViewingKey(viewingKey *ecies.PrivateKey, signerAddress common.Address, viewingPubKeyBytes []byte) {
 	c.viewingPrivKey = viewingKey
-	c.viewingPubKey = viewingPubKeyBytes
-	c.viewingKeyAddr = signerAddress
+
+	c.viewingKeysPrivate[signerAddress] = viewingKey
+	c.viewingKeysPublic[signerAddress] = viewingPubKeyBytes
 }
 
-func (c *ViewingKeyClient) RegisterViewingKey(signature []byte) error {
+func (c *ViewingKeyClient) RegisterViewingKey(signature []byte, signerAddress common.Address) error {
```

Comment:
We need to repass this now, since we don't know which address the signature is for.